### PR TITLE
Front end

### PR DESCRIPTION
**Pull request by MLampa**
We added some front end, pagination, and pg search functions

### CODE REVIEW

File: app/models/venue.rb [hunk 1]
```diff
@@ -18,6 +18,14 @@ class Venue < ActiveRecord::Base
   has_many :reviews
   belongs_to :user
 
+  include PgSearch
+  pg_search_scope :search_by_venue_name,
+                  against: :venue_name,
+                  using: { tsearch: { prefix: true } }
+  # optional multisearch support
+  # multisearchable against: [:venue_name, :street_name, :city, :state, :zip_code]
+  # PgSearch::Multisearch.rebuild(Venue)
```
Comment (davidrf) [inline app/models/venue.rb]: Delete the commented out lines

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,117 +1,110 @@
...
+    <title><%= content_for?(:title) ? yield(:title) : "foundation-rails" %></title>
+
+    <%= stylesheet_link_tag    "application" %>
+    <%= javascript_include_tag "vendor/modernizr" %>
+    <%= javascript_include_tag "application", 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+    <link href="http://cdnjs.cloudflare.com/ajax/libs/foundicons/3.0.0/foundation-icons.css" rel="stylesheet">
+    <link href='https://fonts.googleapis.com/css?family=Righteous' rel='stylesheet' type='text/css'>
+    <link href='https://fonts.googleapis.com/css?family=Open+Sans' rel='stylesheet' type='text/css'>
+    <link href='https://fonts.googleapis.com/css?family=Raleway' rel='stylesheet' type='text/css'>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: add the fonts in a sass file, probably call it `fonts.scss` and do something like [this](http://www.rubyonrailsdeveloper.me/blog/import-google-fonts-in-rails)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,117 +1,110 @@
...
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <title><%= content_for?(:title) ? yield(:title) : "foundation-rails" %></title>
+
+    <%= stylesheet_link_tag    "application" %>
+    <%= javascript_include_tag "vendor/modernizr" %>
+    <%= javascript_include_tag "application", 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+    <link href="http://cdnjs.cloudflare.com/ajax/libs/foundicons/3.0.0/foundation-icons.css" rel="stylesheet">
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: download the foundation icons and put them in a file in your `vendor` folder. Import your this file from you `application.scss`

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,117 +1,110 @@
...
+  $(".right-off-canvas-toggle, .exit-off-canvas").click(function() {
+    $(".top-bar").toggleClass("top-bar-close");
+    $(".middle-bar").toggleClass("middle-bar-close");
+    $(".bottom-bar").toggleClass("bottom-bar-close");
   });
-  </script>
-</body>
-</html>
+});
+</script>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Put this javascript in your asset pipeline

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,117 +1,110 @@
...
-  <ul class="inline-list social">
-    <li><i class="fi-social-facebook"></i></li>
-    <li><i class="fi-social-pinterest"></i></li>
-    <li><i class="fi-social-youtube"></i></li>
-    <li><i class="fi-social-twitter"></i></li>
-  </ul>
-</footer>
+    Follow us:
+      <div class="twitter"><a href="https://www.twitter.com"><img src="https://lh3.ggpht.com/lSLM0xhCA1RZOwaQcjhlwmsvaIQYaP3c5qbDKCgLALhydrgExnaSKZdGa8S3YtRuVA=w300"></a></div>
+  </footer>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract footer into a partial `views/applications/_footer.html.erb`

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -1,117 +1,110 @@
...
+            <ul class="side-nav main">
+              <li><a href="#">bowling</a></li>
+              <li><a href="#">basketball</a></li>
+            </ul>
+        </aside>
+
+      <div class="row">
+        <% flash.each do |key, value| %>
+          <div class="flash <%= key %>"><%= value %></div>
         <% end %>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract flash messages into a partial `views/application/_flash.html.erb`

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -1,117 +1,110 @@
...
+    <div class="off-canvas-wrap" data-offcanvas>
+      <div class="inner-wrap">
+        <a class="right-off-canvas-toggle" href="#" >
+          <div class="menu">
+            <div class="off-canvas-bar"></div>
+            <div class="off-canvas-middle-bar"></div>
+            <div class="off-canvas-bottom-bar"></div>
+          </div>
+        </a>
+      </nav>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Delete line 73, this element does not have an opening tag.

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -1,117 +1,110 @@
...
+  <body>
+    <div class="off-canvas-wrap" data-offcanvas>
+      <div class="inner-wrap">
+        <a class="right-off-canvas-toggle" href="#" >
+          <div class="menu">
+            <div class="off-canvas-bar"></div>
+            <div class="off-canvas-middle-bar"></div>
+            <div class="off-canvas-bottom-bar"></div>
+          </div>
+        </a>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract flash messages into a partial `views/application/_right_off_canvas_toggle.html.erb`

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -1,117 +1,110 @@
...
+      </nav>
+        <!-- Off Canvas Menu -->
+        <aside class="right-off-canvas-menu">
+            <!-- whatever you want goes here -->
+            <ul class="side-nav main">
+              <li><a href="#">bowling</a></li>
+              <li><a href="#">basketball</a></li>
+            </ul>
+        </aside>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract lines 75-81 into a partial `views/application/_off_canvas_menu.html.erb`

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -1,117 +1,110 @@
...
+      <div class="inner-wrap">
+        <a class="right-off-canvas-toggle" href="#" >
+          <div class="menu">
+            <div class="off-canvas-bar"></div>
+            <div class="off-canvas-middle-bar"></div>
+            <div class="off-canvas-bottom-bar"></div>
+          </div>
+        </a>
+      </nav>
+        <!-- Off Canvas Menu -->
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Delete this line

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -1,117 +1,110 @@
...
+        <aside class="right-off-canvas-menu">
+            <!-- whatever you want goes here -->
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Delete this line

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -1,117 +1,110 @@
...
+              <li><%= link_to "Profile", "/users/#{current_user.id}" %></li>
+              <li><%= link_to "Sign Out", destroy_user_session_path, :method => :delete %></li>
+              <% else %>
+              <li><%= link_to "Sign In", new_user_session_path %></li>
+              <% end %>
+            </ul>
+          </section>
+        </nav>
+      </div>
+    </div>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract lines 17-60 into a partial `views/application/_top_bar.html.erb`

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -1,117 +1,110 @@
...
+                  <%= form_tag("/search", method: "get") do %>
+                    <div class="large-8 small-9 columns">
+                      <%= text_field_tag :query, params[:query], attr: "Find Stuff" %>
+                    </div>
+                    <div class="large-4 small-3 columns">
+                      <%= submit_tag "Search", name: nil %>
+                    </div>
+                  <% end %>
+                </div>
+              </li>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract lines 28-39 into a partial `views/application/_top_bar_form.html.erb` You can render this partial with 

```
<%= render "top_bar_form" %>
```

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,117 +1,110 @@
...
+              <li class="has-dropdown">
+                <a href="#">Filter by Boston Neighborhood</a>
+                <ul class="dropdown">
+                  <li><a href="#">Back Bay</a></li>
+                  <li><a href="#">Central Square</a></li>
+                </ul>
+              </li>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract lines 40-46 into a partial `views/application/_top_bar_filter.html.erb`

File: app/views/layouts/application.html.erb [hunk 15]
```diff
@@ -1,117 +1,110 @@
...
+              <% if user_signed_in? %>
+                <% unless current_user.avatar.url.nil? %>
+                  <li class="picture"><%= image_tag(current_user.avatar) %></li>
+                <% end %>
+              <li><%= link_to "Profile", "/users/#{current_user.id}" %></li>
+              <li><%= link_to "Sign Out", destroy_user_session_path, :method => :delete %></li>
+              <% else %>
+              <li><%= link_to "Sign In", new_user_session_path %></li>
+              <% end %>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract lines 47-55 into a partial `views/application/_top_bar_authentication.html.erb`

File: app/views/layouts/application.html.erb [hunk 16]
```diff
@@ -1,117 +1,110 @@
...
+    <link href='https://fonts.googleapis.com/css?family=Raleway' rel='stylesheet' type='text/css'>
+
+    <div data-sticky-container>
+      <div class="sticky" data-sticky data-margin-top=0 style="width:100%";>
+        <nav class="top-bar" data-topbar role="navigation">
+          <ul class="title-area">
+            <li class="name">
+              <h1><a href="/" style="font-family: 'Righteous', cursive; font-size: 25px">Jump<u>e</u>r</a></h1>
+            </li>
           </ul>
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract lines 20-24 into a partial `views/application/_top_bar_title.html.erb` You can render this partial with

```
<%= render "top_bar_title" %>
```

File: app/views/layouts/application.html.erb (hunk 8 above)
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Extract lines 66-72 into a partial `views/application/_off_canvas_toggle.html.erb`

File: config/initializers/kaminari_config.rb [hunk 17]
```diff
@@ -0,0 +1,10 @@
+Kaminari.configure do |config|
+  config.default_per_page = 6
+  # config.max_per_page = nil
+  # config.window = 4
+  # config.outer_window = 0
+  # config.left = 0
+  # config.right = 0
+  # config.page_method_name = :page
+  # config.param_name = :page
```
Comment (davidrf) [inline config/initializers/kaminari_config.rb]: Delete commented out lines

File: app/views/venues/show.html.erb [hunk 18]
```diff
@@ -1,36 +1,43 @@
...
+        </div>
+        <div class="not-helpful box"><%= review.revdown_count %></div>
+      </div>
+
+      <% if review.deletable_by(current_user) %>
+        <%= button_to "Delete Review", venue_review_path(@venue, review), class: "button", method: "delete", label: "Delete Review" %>
+      <% end %>
+      <% if review.deletable_by(current_user) %>
+        <%= button_to "Edit Review", edit_venue_review_path(@venue, review), class: "button", method: "get", label: "Edit Review" %>
+      <% end %>
```
Comment (davidrf) [inline app/views/venues/show.html.erb]: Wrap both these buttons in one `if` statement

File: app/views/venues/show.html.erb [hunk 19]
```diff
@@ -1,36 +1,43 @@
...
-<% end %>
-<% if @venue.deletable_by(current_user) %>
-  <%= button_to "Edit Venue", edit_venue_path, class: "button", method: "get", label: "Edit Venue"%>
-<% end %>
+  <% if @venue.deletable_by(current_user) %>
+    <%= button_to "Delete Venue", venue_path, class: "button", method: "delete", label: "Delete Venue"%>
+  <% end %>
+  <% if @venue.deletable_by(current_user) %>
+    <%= button_to "Edit Venue", edit_venue_path, class: "button", method: "get", label: "Edit Venue"%>
+  <% end %>
```
Comment (davidrf) [inline app/views/venues/show.html.erb:8]: Wrap both buttons in one `if` statement

File: app/views/venues/index.html.erb [hunk 20]
```diff
@@ -1,21 +1,42 @@
...
+        </div>
       <% end %>
     </div>
+    <% if @reviews %>
+      <div class="floating-container">
+        <% @reviews.each do |review| %>
+          </h4><%= link_to review.review_body, venue_path(review) %></h4>
+        <% end %>
+      </div>
+    <% end %>
```
Comment (davidrf) [inline app/views/venues/index.html.erb]: Why do you have reviews in your venue index???? You should probably remove this.

File: app/views/venues/index.html.erb [hunk 21]
```diff
@@ -1,21 +1,42 @@
...
+
+            <div class="panel">
+              <h6 class="subheader"><%= venue.description %></h6>
+              <div class="image-overlay-content">
+                <h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+              </div>
+            </div>
+          </div>
+        </div>
       <% end %>
```
Comment (davidrf) [inline app/views/venues/index.html.erb]: Move lines 12-23 into `views/venues/_venue.html.erb`. Then remove the `each` block. You can now just do `<%= @venues %>`

File: app/views/venues/show.html.erb [hunk 22]
```diff
@@ -1,36 +1,43 @@
...
+      </div>
+
+      <% if review.deletable_by(current_user) %>
+        <%= button_to "Delete Review", venue_review_path(@venue, review), class: "button", method: "delete", label: "Delete Review" %>
+      <% end %>
+      <% if review.deletable_by(current_user) %>
+        <%= button_to "Edit Review", edit_venue_review_path(@venue, review), class: "button", method: "get", label: "Edit Review" %>
+      <% end %>
+    </div>
   </div>
```
Comment (davidrf) [inline app/views/venues/show.html.erb]: Move lines 14-38 into `views/reviews/_review.html.erb` then remove the `each` block and just do `<%= @reviews %>`

File: app/views/layouts/application.html.erb [hunk 23]
```diff
@@ -1,117 +1,36 @@
...
+    <%= render "layouts/off_canvas_menu_end" %>
+
+    <script>
+    $(function() {
+      $(".right-off-canvas-toggle, .exit-off-canvas").click(function() {
+        $(".top-bar").toggleClass("top-bar-close");
+        $(".middle-bar").toggleClass("middle-bar-close");
+        $(".bottom-bar").toggleClass("bottom-bar-close");
+      });
     });
```
Comment (davidrf) [inline app/views/layouts/application.html.erb]: Move this javascript to a `js` file in your asset pipeline.

File: app/views/venues/index.html.erb [hunk 24]
```diff
@@ -1,14 +1,15 @@
...
+  <div class="row">
+    <div class="small-12 medium-10 medium-centered large-8 large-centered columns">
+
+      <div class="floating-container">
+        <div class="row venue-header">
+          <h1>Venues</h1>
+        </div>
+      </div>
+
+      <%= render "venues/venue" %>
```
Comment (davidrf) [inline app/views/venues/index.html.erb]: `<%= render @venues %>`

File: app/views/venues/_venue.html.erb [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+
+    <div class="panel">
+      <h6 class="subheader"><%= venue.description %></h6>
+      <div class="image-overlay-content">
+        <h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+      </div>
+    </div>
+  </div>
+</div>
+<% end %>
```
Comment (davidrf) [inline app/views/venues/_venue.html.erb]: delete this line

File: app/views/venues/_venue.html.erb [hunk 26]
```diff
@@ -0,0 +1,14 @@
+<% @venues.each do |venue| %>
```
Comment (davidrf) [inline app/views/venues/_venue.html.erb]: delete this line

File: app/views/venues/_reviews.html.erb [hunk 27]
```diff
@@ -0,0 +1,27 @@
+<% @reviews.each do |review| %>
```
Comment (davidrf) [inline app/views/venues/_reviews.html.erb]: Remove this line

File: app/views/venues/_reviews.html.erb [hunk 28]
```diff
@@ -0,0 +1,27 @@
...
+
+    <% if review.deletable_by(current_user) %>
+      <%= button_to "Delete Review", venue_review_path(@venue, review), class: "button", method: "delete", label: "Delete Review" %>
+    <% end %>
+    <% if review.deletable_by(current_user) %>
+      <%= button_to "Edit Review", edit_venue_review_path(@venue, review), class: "button", method: "get", label: "Edit Review" %>
+    <% end %>
+  </div>
+</div>
+<% end %>
```
Comment (davidrf) [inline app/views/venues/_reviews.html.erb]: Remove this line

File: app/views/venues/_reviews.html.erb (hunk 27 above)
Comment (davidrf) [inline app/views/venues/_reviews.html.erb]: Also, move this file to `app/views/reviews/`

File: app/views/venues/show.html.erb [hunk 29]
```diff
@@ -1,36 +1,16 @@
...
 
-<% if @venue.deletable_by(current_user) %>
-  <%= button_to "Delete Venue", venue_path, class: "button", method: "delete", label: "Delete Venue"%>
-<% end %>
-<% if @venue.deletable_by(current_user) %>
-  <%= button_to "Edit Venue", edit_venue_path, class: "button", method: "get", label: "Edit Venue"%>
-<% end %>
+  <% if @venue.deletable_by(current_user) %>
+    <%= button_to "Delete Venue", venue_path, class: "button", method: "delete", label: "Delete Venue"%>
+  <% elsif @venue.deletable_by(current_user) %>
```
Comment (davidrf) [inline app/views/venues/show.html.erb]: Delete this line

File: app/views/venues/show.html.erb [hunk 30]
```diff
@@ -1,36 +1,16 @@
...
-  </div>
-  </div>
-  <% if review.deletable_by(current_user) %>
-  <%= button_to "Delete Review", venue_review_path(@venue, review), class: "button", method: "delete", label: "Delete Review" %>
-  <% end %>
-  <% if review.deletable_by(current_user) %>
-  <%= button_to "Edit Review", edit_venue_review_path(@venue, review), class: "button", method: "get", label: "Edit Review" %>
-  <% end %>
-  <% end %>
+  <%= render "venues/reviews" %>
```
Comment (davidrf) [inline app/views/venues/show.html.erb]: `<%= render @reviews %>`

File: app/views/venues/_reviews.html.erb [hunk 31]
```diff
@@ -0,0 +1,27 @@
+<% @reviews.each do |review| %>
+<div class="row">
```
Comment (davidrf) [inline app/views/venues/_reviews.html.erb]: Also, move this file to `app/views/reviews/`

File: app/views/search/index.html.erb [hunk 32]
```diff
@@ -0,0 +1,23 @@
...
+      <div class="floating-container">
+        <% @venues.each do |venue| %>
+          <h4><%= link_to venue.venue_name, venue_path(venue) %></h4>
+        <% end %>
+      </div>
+    <% end %>
+    <% if @reviews %>
+      <div class="floating-container">
+        <% @reviews.each do |review| %>
+          <h4><%= link_to review.review_body, venue_path(review) %></h4>
```
Comment (davidrf) [inline app/views/search/index.html.erb]: Pretty sure you mean `venue_path(review.venue)`

File: app/views/layouts/_off_canvas_menu_end.html.erb [hunk 33]
```diff
@@ -0,0 +1,3 @@
+    <a class="exit-off-canvas"></a>
+  </div>
+</div>
```
Comment (davidrf) [inline app/views/layouts/_off_canvas_menu_end.html.erb]: Delete this partial and put the html back in `application.html.erb`

File: app/views/layouts/_off_canvas_menu_start.html.erb [hunk 34]
```diff
@@ -0,0 +1,17 @@
+<div class="off-canvas-wrap" data-offcanvas>
+  <div class="inner-wrap">
```
Comment (davidrf) [inline app/views/layouts/_off_canvas_menu_start.html.erb]: Put back the first two lines of this file back in `application.html.erb` and rename this file to `_off_canvas_menu.html.erb`

File: config/initializers/kaminari_config.rb [hunk 35]
```diff
@@ -0,0 +1,10 @@
+Kaminari.configure do |config|
```
Comment (davidrf) [inline config/initializers/kaminari_config.rb:1]: Go home hound you're drunk

Comment (MLampa) [inline config/initializers/kaminari_config.rb:1]: lol how did I just see this now... lol brilliant note, D-Rod lol

**Pull request merged**